src/database/sqlite.rs

```diff
@@ -42,7 +42,7 @@ static MIGRATIONS: &[&str] = &[
     "CREATE TABLE transaction_details (txid BLOB, timestamp INTEGER, received INTEGER, sent INTEGER, fee INTEGER, height INTEGER);",
     "INSERT INTO transaction_details SELECT txid, timestamp, received, sent, fee, height FROM transaction_details_old;",
     "DROP TABLE transaction_details_old;",
-    "ALTER TABLE utxos ADD COLUMN is_spent;",
+    "ALTER TABLE utxos ADD COLUMN is_spent BOOLEAN CHECK (is_spent IN (0, 1));",
```

Comment:
The problem with this approach is that databases that already did this migration won't be changed (see the migration code: https://github.com/bitcoindevkit/bdk/blob/master/src/database/sqlite.rs#L953-L976)

Maybe it's better to add YA migration altering the utxo table? Idk if it's possible to change the type of a column

Reply:
Yes it's probably better to add this as an extra migration

Reply:
@danielabrozzoni @afilini, it is not possible to modify an existing column datatype in sqlite. So using something like `ALTER COLUMN` will not work here. Another option will be to rename the table to `table_name_old` and copy all its content to the new table. Here is an example from the migration code:
```sql
CREATE TABLE transaction_details (txid BLOB, timestamp INTEGER, received INTEGER, sent INTEGER, fee INTEGER, height INTEGER, verified INTEGER DEFAULT 0);
ALTER TABLE transaction_details RENAME TO transaction_details_old;
CREATE TABLE transaction_details (txid BLOB, timestamp INTEGER, received INTEGER, sent INTEGER, fee INTEGER, height INTEGER);
INSERT INTO transaction_details SELECT txid, timestamp, received, sent, fee, height FROM transaction_details_old;
DROP TABLE transaction_details_old;
```
    
   This is a lot of gymnastics for a simple datatype change. Another possibility will be to allow the current implementation. What do you guys think?

Reply:
I'd honestly go with renaming the old table and moving everything to a new one... It is a bit of gymnastic, but it's important that our type remain strict for everyone, not only for new users of bdk

Reply:
@danielabrozzoni , updated following your feedback.